### PR TITLE
Implement `schemas.patch` RPC method

### LINKED ISSUE #3627: Implement `schemas.patch` RPC method

### PR DESCRIPTION
**Pull request by seancolsen**
Fixes #3627

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: db/sql/00_msar.sql [hunk 1]
```diff
@@ -834,95 +842,71 @@ $$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
...
+CREATE OR REPLACE FUNCTION msar.set_schema_description(
+  sch_id oid,
+  description text
+) RETURNS void AS $$/*
+Set the PostgreSQL description (aka COMMENT) of a schema.
 
-CREATE OR REPLACE FUNCTION
-__msar.comment_on_schema(sch_name text, comment_ text) RETURNS TEXT AS $$/*
-Change the description of a schema, returning command executed.
+Descriptions are removed by passing an empty string. Passing a NULL description will cause
```
Comment (mathemancer) [inline db/sql/00_msar.sql:871]: Huh. Apparently `NULL` and empty string descriptions are equivalent as far as PostgreSQL is concerned. I didn't know that.

File: db/sql/00_msar.sql [hunk 2]
```diff
@@ -834,95 +842,71 @@ $$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
...
+this function to return NULL withou doing anything.
```
Comment (mathemancer) [inline db/sql/00_msar.sql:872]:
```suggestion
this function to return NULL without doing anything.
```

File: db/sql/00_msar.sql [hunk 3]
```diff
@@ -834,95 +842,71 @@ $$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
...
 
-CREATE OR REPLACE FUNCTION msar.comment_on_schema(sch_id oid, comment_ text) RETURNS TEXT AS $$/*
-Change the description of a schema, returning command executed.
+CREATE OR REPLACE FUNCTION msar.patch_schema(sch_name text, patch jsonb) RETURNS void AS $$/*
+Modify a schema according to the given patch.
 
 Args:
-  sch_id: The OID of the schema.
-  comment_: The new comment, UNQUOTED
+  sch_id: The name of the schema.
```
Comment (mathemancer) [inline db/sql/00_msar.sql:905]:
```suggestion
  sch_name: The name of the schema, UNQUOTED.
```

File: db/sql/test_00_msar.sql [hunk 4]
```diff
@@ -1310,47 +1310,40 @@ END;
...
-  PERFORM __setup_alter_schema();
-  PERFORM msar.rename_schema(
-    sch_id => 'alter_me'::regnamespace::oid,
-    new_sch_name => 'altered'
-  );
-  RETURN NEXT hasnt_schema('alter_me');
-  RETURN NEXT has_schema('altered');
-END;
-$$ LANGUAGE plpgsql;
+  -- Edge case: setting the description to null doesn't actually remove it. This behavior is
```
Comment (mathemancer) [inline db/sql/test_00_msar.sql:1328]: I think it's fine. At some point, we should put some effort into making this behavior consistent across the code base.

**Review by mathemancer — APPROVED** [review]
I found a couple of documentation typos. Feel free to merge once those are fixed.

File: db/sql/00_msar.sql (hunk 3 above)
Comment (seancolsen) [reply db/sql/00_msar.sql]: Fixed in 1c1b55a287fdae83fc1856d9ab48201cecc848f2

**Pull request merged**